frame/undercollateralized-loans/src/helpers.rs

```diff
@@ -0,0 +1,529 @@
+	}
+
+	// Check if provided account id belongs to the market manager.
+	pub(crate) fn is_market_manager_account(
+		account_id: &T::AccountId,
+		market_account_id_ref: &T::AccountId,
+	) -> Result<bool, DispatchError> {
+		let market_config = Self::get_market_config_via_account_id(market_account_id_ref)?;
+		Ok(market_config.manager() == account_id)
+	}
+
+	// TODO: @mikolaichuk: Add weights calculation
+	// #generalization
+	pub(crate) fn treat_vaults_balance(block_number: T::BlockNumber) -> () {
+		let _ = with_transaction(|| {
+			let now = Self::now();
+			let mut errors = crate::MarketsStorage::<T>::iter()
+				.map(|(market_account_id, market_info)| {
+					match Self::available_funds(&market_info.config(), &market_account_id)? {
+						FundsAvailability::Withdrawable(balance) => {
+							Self::handle_withdrawable(
+								&market_info.config(),
+								&market_account_id,
+								balance,
+							)?;
+						},
+						FundsAvailability::Depositable(balance) => {
+							Self::handle_depositable(
+								&market_info.config(),
+								&market_account_id,
+								balance,
+							)?;
+						},
+						FundsAvailability::MustLiquidate => {
+							Self::handle_must_liquidate(&market_info.config(), market_account_id)?;
+						},
+					}
+					Result::<(), DispatchError>::Ok(())
+				})
+				.filter_map(|r| match r {
+					Ok(_) => None,
+					Err(err) => Some(err),
+				})
+				.peekable();
+
+			if errors.peek().is_none() {
+				TransactionOutcome::Commit(Ok(1000))
+			} else {
+				errors.for_each(|e| {
+					log::error!(
+						"This should never happen, could not initialize block!!! {:#?} {:#?}",
+						block_number,
+						e
+					)
+				});
+				TransactionOutcome::Rollback(Err(DispatchError::Other(
+					"failed to initialize block",
+				)))
+			}
+		});
+	}
+
+	pub(crate) fn check_payments(today: Timestamp) -> () {
+		// Retrive collection of loans(loans' accounts ids) which have to be paid now.
+		// If nothing found we get empty set.
+		let loans_accounts_ids = crate::ScheduleStorage::<T>::try_get(today).unwrap_or_default();
+
+		for loan_account_id in loans_accounts_ids {
+			// Treat situation when non-existent loan's id is in global schedule.
+			let loan_config = match Self::get_loan_config_via_account_id(&loan_account_id) {
+				Ok(loan_config) => loan_config,
+				Err(_) => continue,
+			};
+			match Self::check_payment(&loan_config, today) {
+				RepaymentResult::Failed(error) => log::error!(
+					"Payment check for loan {:?} were failed becasue of the following error: {:?}",
+					loan_account_id,
+					error
+				),
+				RepaymentResult::PaymentMadeOnTime => (),
+				RepaymentResult::LastPaymentMadeOnTime =>
+					Self::close_loan_contract(&loan_config, false),
+				RepaymentResult::PaymentIsOverdue => Self::do_liquidate(&loan_config),
+			}
+			// We do not need information regarding this date anymore.
+			crate::ScheduleStorage::<T>::remove(today);
+		}
+	}
+
+	// Check if a payment is succeed.
```

Comment:
NIT: "Check if a payment is succeed" -> "Check if a payment is successful."

Reply:
fixed